### PR TITLE
Sample application magic link example

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
This change is dependent on pending work being done on https://github.com/okta/okta-core/pull/61907

- OKTA-448556 - different browser
- OKTA-448553 - same browser

https://user-images.githubusercontent.com/63638027/152814607-28873237-c06b-43c6-9cdb-4cf72afd197b.mp4

### CODE REVIEW

**Review by laura-rodriguez — COMMENTED** [review]
This LGTM 👍 
I'd like to see a demo once the API is fixed.

**Review by andriizhegurov-okta — APPROVED** [review]
👍

File: samples/samples-aspnet/embedded-auth-with-sdk/embedded-auth-with-sdk/Controllers/AccountController.cs [hunk 1]
```diff
@@ -68,6 +68,7 @@ public async Task<ActionResult> LoginAsync(LoginViewModel model)
             {
                 var authnResponse = await _idxClient.AuthenticateAsync(authnOptions).ConfigureAwait(false);
                 Session["idxContext"] = authnResponse.IdxContext;
+                Session[authnResponse?.IdxContext.State] = authnResponse?.IdxContext; // save context in session keyed by state for retrieval by magiclink, see MagicLinkController.Callback.
```
Comment (danielmaharry-okta) [inline samples/samples-aspnet/embedded-auth-with-sdk/embedded-auth-with-sdk/Controllers/AccountController.cs:71]: This line needs to read

`Session[authnResponse?.IdxContext?.State] = authnResponse?.IdxContext;`

ie Add ? to authnResponse?.IdxContext

Comment (bryanapellanes-okta) [reply samples/samples-aspnet/embedded-auth-with-sdk/embedded-auth-with-sdk/Controllers/AccountController.cs:71]: that said if authnResponse is null we've got problems as well... 🤷 doesn't hurt to add the null check.

**Pull request merged**